api/submission/fetch_course_report_test.go

```diff
@@ -0,0 +1,110 @@
+package submission
+
+import (
+	"reflect"
+	"testing"
+
+    "github.com/edulinq/autograder/common"
+	"github.com/edulinq/autograder/api/core"
+	"github.com/edulinq/autograder/model"
+	"github.com/edulinq/autograder/report"
+	"github.com/edulinq/autograder/util"
+)
+
+func TestCourseReport(test *testing.T) {
+    testCases := []struct{
+        role model.UserRole
+        courseId string 
+        coursError bool
+        permError bool
+        result *FetchCourseReportResponse
+    }{
+        // Admin Valid Course
```

Comment:
I think you mean "Admin, Valid Course".
We want to try and keep our comments as grammatically correct as possible.